### PR TITLE
Fix packaging and documentation

### LINKED ISSUE #96: Installation problems on virtual environments with source releases
**Describe the bug**
I tried `pip install .` but not all subpackages are installed, for example `d3rlpy.models.torch.q_functions` (see https://github.com/readthedocs/readthedocs.org/issues/8326).

**To Reproduce**
1. Create virtual environment and activate it: `python3.7 -m venv .venv && source .venv/bin/activate`
2. Install build dependencies: `pip install numpy cython`
1. Run `pip install .`
2. Change to a different directory (to avoid importing from the same directory, see https://hynek.me/articles/testing-packaging/#src)
3. Run `import d3rlpy`
4. See crash

**Expected behavior**
The package source code can be installed and properly imported in a virtual environment.

### PR DESCRIPTION
**Pull request by astrojuanlu**
Fix #96

### CODE REVIEW

**Comment by astrojuanlu** [comment]
Also, you can enable the PR builder on Read the Docs to see how this would look like: https://blog.readthedocs.com/pull-request-builder-general-availability/

**Comment by takuseno** [comment]
@astrojuanlu Thanks! It's enabled now!

**Comment by astrojuanlu** [comment]
The Read the Docs build is done now! Still fails importing, but as I said it looks like a problem with the aurosummary templates

**Comment by astrojuanlu** [comment]
As I explained in https://github.com/sphinx-doc/sphinx/issues/7989, the `WARNING: [autosummary] failed to import 'd3rlpy.algos.AWAC': no module named d3rlpy.algos.AWAC` doesn't actually mean that the `d3rlpy.algos.AWAC` module is being imported. Rather, it is masking a deeper error:

```
Traceback (most recent call last):
  File "/home/docs/checkouts/readthedocs.org/user_builds/d3rlpy/envs/latest/lib/python3.7/site-packages/sphinx/ext/autodoc/importer.py", line 70, in import_module
    return importlib.import_module(modname)
  File "/home/docs/checkouts/readthedocs.org/user_builds/d3rlpy/envs/latest/lib/python3.7/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1006, in _gcd_import
  File "<frozen importlib._bootstrap>", line 983, in _find_and_load
  File "<frozen importlib._bootstrap>", line 953, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "<frozen importlib._bootstrap>", line 1006, in _gcd_import
  File "<frozen importlib._bootstrap>", line 983, in _find_and_load
  File "<frozen importlib._bootstrap>", line 967, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 677, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 728, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/src/docs/../d3rlpy/__init__.py", line 6, in <module>
    from . import (
  File "/src/docs/../d3rlpy/algos/__init__.py", line 3, in <module>
    from .awac import AWAC
  File "/src/docs/../d3rlpy/algos/awac.py", line 3, in <module>
    from ..argument_utility import (
  File "/src/docs/../d3rlpy/argument_utility.py", line 6, in <module>
    from .models.encoders import EncoderFactory, create_encoder_factory
  File "/src/docs/../d3rlpy/models/__init__.py", line 1, in <module>
    from . import encoders, optimizers, q_functions
  File "/src/docs/../d3rlpy/models/encoders.py", line 7, in <module>
    from ..torch_utility import Swish
  File "/src/docs/../d3rlpy/torch_utility.py", line 10, in <module>
    from .dataset import TransitionMiniBatch
ModuleNotFoundError: No module named 'd3rlpy.dataset'
```

However, launching a Python interpreter doesn't have this problem. I think I got how to fix it, will push one more commit.

**Comment by astrojuanlu** [comment]
@takuseno problem fixed, have a look at the build logs 👍🏽

**Comment by astrojuanlu** [comment]
Rendered version:  https://d3rlpy--97.org.readthedocs.build/en/97/references/algos.html#continuous-control-algorithms

**Review by takuseno — CHANGES_REQUESTED** [review]
Thanks for the change! You save my days!!! I've left one minor comment. I'd like you to take a look at it. Thanks!

File: setup.py [hunk 1]
```diff
@@ -62,21 +62,7 @@
...
-                    "d3rlpy.iterators",
-                    "d3rlpy.metrics",
-                    "d3rlpy.models",
-                    "d3rlpy.models.torch",
-                    "d3rlpy.online",
-                    "d3rlpy.ope",
-                    "d3rlpy.ope.torch",
-                    "d3rlpy.preprocessing",
-                    "d3rlpy.wrappers"],
+          packages=find_packages(),
```
[the remark this answers is not in the document]
Comment (astrojuanlu) [reply setup.py:65]: Or, we can `exclude="tests"` :) On it

**Review by takuseno — APPROVED** [review]
Thanks for the change! I'll merge this when the test has passed.

**Pull request merged**